File: 882-peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var peakIndexInMountainArray = function(arr) {
     var ind = 0;
        var maxi = 0;
        for (var i = 1; i < arr.length - 1; i++) {
            var prev = arr[i - 1];
            var next = arr[i + 1];
            var curr = arr[i];
            if (curr > prev && curr > next) {
                maxi = Math.max(maxi, curr);
                ind = i;
            }
        }
        return ind;
};

File: 882-peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int ind = 0;
        int maxi = 0;
        for (int i = 1; i < arr.size() - 1; i++) {
            int prev = arr[i - 1];
            int next = arr[i + 1];
            int curr = arr[i];
            if (curr > prev && curr > next) {
                maxi = max(maxi, curr);
                ind = i;
            }
        }
        return ind;
    }
};
